### PR TITLE
Android Build Fixes

### PR DESCRIPTION
**Pull request by devshgraphicsprogramming**
## Description
Changes to make Nabla run on Android

## Testing 
Build BaW with this

## TODO list:

- [ ] Undo the changes to surfaces (nice new templates and hidden headers)
- [ ] Stop globbing and installing headers in `src`

### CODE REVIEW

File: include/nbl/video/surface/CSurfaceVulkan.h [hunk 1]
```diff
@@ -2,72 +2,198 @@
...
-class ISurfaceVulkan : public ISurface
+template <typename Window>
+class CSurfaceVulkan final : public CSurface<Window>
 {
-    using base_t = ISurface;
 public:
-    bool isSupportedForPhysicalDevice(const IPhysicalDevice* dev, uint32_t _queueFamIx) const override;
-    void getAvailableFormatsForPhysicalDevice(const IPhysicalDevice* physicalDevice, uint32_t& formatCount, ISurface::SFormat* formats) const override;
-    E_PRESENT_MODE getAvailablePresentModesForPhysicalDevice(const IPhysicalDevice* physicalDevice) const override;
-    bool getSurfaceCapabilitiesForPhysicalDevice(const IPhysicalDevice* physicalDevice, ISurface::SCapabilities& capabilities) const override;
```
Comment (devshgraphicsprogramming) [inline include/nbl/video/surface/CSurfaceVulkan.h:22]: restore this

File: include/nbl/video/surface/CSurfaceVulkan.h [hunk 2]
```diff
@@ -2,72 +2,198 @@
...
+    }
 
-#ifdef _NBL_PLATFORM_WINDOWS_
-class CSurfaceVulkanWin32 final : public CSurface<ui::IWindowWin32, ISurfaceVulkan>
-{
-    using this_t = CSurfaceVulkanWin32;
-    using base_t = CSurface<ui::IWindowWin32, ISurfaceVulkan>;
-public:
-    CSurfaceVulkanWin32(core::smart_refctd_ptr<ui::IWindowWin32>&& window, core::smart_refctd_ptr<IAPIConnection>&& api, VkSurfaceKHR surf) :
-        base_t(std::move(window), std::move(api), surf)
```
Comment (devshgraphicsprogramming) [inline include/nbl/video/surface/CSurfaceVulkan.h:40]: restore this

File: include/nbl/video/surface/ISurface.h [hunk 3]
```diff
@@ -109,8 +109,8 @@ class ISurface : public core::IReferenceCounted
         virtual const void* getNativeWindowHandle() const = 0;
 };
 
-template<class Window, class ImmediateBase = ISurface>
-class CSurface : public ImmediateBase
```
Comment (devshgraphicsprogramming) [inline include/nbl/video/surface/ISurface.h:113]: restore this

File: src/nbl/CMakeLists.txt [hunk 4]
```diff
@@ -176,6 +176,19 @@ file(GLOB_RECURSE TEMP_GLOB_RES "${NBL_ROOT_PATH}/include/*.h")
...
+foreach(NBL_CURRENT_HEADER IN LISTS NABLA_HEADERS_PUBLIC2)
+	cmake_path(GET NBL_CURRENT_HEADER PARENT_PATH NBL_CURRENT_HEADER_DIR)
+	file(RELATIVE_PATH NBL_TMP_REL_DESTINATION "${NBL_ROOT_PATH}/src" ${NBL_CURRENT_HEADER_DIR})
+	
+	install(FILES ${NBL_CURRENT_HEADER} DESTINATION include/${NBL_TMP_REL_DESTINATION} CONFIGURATIONS Release)
+	install(FILES ${NBL_CURRENT_HEADER} DESTINATION debug/include/${NBL_TMP_REL_DESTINATION} CONFIGURATIONS Debug)
+	install(FILES ${NBL_CURRENT_HEADER} DESTINATION relwithdebinfo/include/${NBL_TMP_REL_DESTINATION} CONFIGURATIONS RelWithDebInfo)
+endforeach()
+
+unset(NABLA_HEADERS_PUBLIC2 ${NBL_TMP_FULL_PATHS})
```
Comment (devshgraphicsprogramming) [inline src/nbl/CMakeLists.txt:191]: undo this

File: src/nbl/CMakeLists.txt [hunk 5]
```diff
@@ -675,6 +687,12 @@ nbl_install_headers("${NABLA_HEADERS_TO_INSTALL}" "${NBL_ROOT_PATH}/include")
 nbl_install_headers("${NABLA_HEADERS_TO_INSTALL_SOURCE_REFACTOR}" "${NBL_ROOT_PATH}/source/Nabla")
 nbl_install_config_header(BuildConfigOptions.h)
 
+macro(nbl_install_program _TRGT)
+	install(PROGRAMS $<TARGET_FILE:${_TRGT}> DESTINATION lib CONFIGURATIONS Release)
+	install(PROGRAMS $<TARGET_FILE:${_TRGT}> DESTINATION debug/lib CONFIGURATIONS Debug)
+	install(PROGRAMS $<TARGET_FILE:${_TRGT}> DESTINATION relwithdebinfo/lib CONFIGURATIONS RelWithDebInfo)
+endmacro()
```
Comment (devshgraphicsprogramming) [inline src/nbl/CMakeLists.txt:694]: why is it install program and not install lib?

File: src/nbl/video/CSurfaceVulkan.cpp [hunk 6]
```diff
@@ -1,121 +0,0 @@
...
-		{
-			return false;
-		}
-	}
-	void ISurfaceVulkan::getAvailableFormatsForPhysicalDevice(const IPhysicalDevice* physicalDevice, uint32_t& formatCount, ISurface::SFormat* formats) const
-	{
-		constexpr uint32_t MAX_SURFACE_FORMAT_COUNT = 1000u;
-
-		if (physicalDevice && physicalDevice->getAPIType() != EAT_VULKAN)
-			return;
```
Comment (devshgraphicsprogramming) [inline src/nbl/video/CSurfaceVulkan.cpp:29]: restore the file

File: src/nbl/CMakeLists.txt (hunk 5 above)
Comment (AnastaZIuk) [reply src/nbl/CMakeLists.txt:694]: because shared objects are `PROGRAMS` in terms of CMake terminology

Comment (devshgraphicsprogramming) [reply src/nbl/CMakeLists.txt:694]: XD

**Comment by devshgraphicsprogramming** [comment]
all the stuff is already in` v1.x`

**Pull request closed without merging**